app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt

```diff
@@ -195,4 +204,46 @@ object EventUtils {
             return ""
         }
     }
+
+    /**
+     *  share event detail along with event image
+     *  if image loading is successful then imageView tag will be set to String
+     *  So if imageView tag is not null then share image else only event details
+     *  @param eventImage to extract the image and use context
+     *
+     */
+    fun share(event: Event, eventImage: ImageView) {
+        val sendIntent = Intent()
+        sendIntent.action = Intent.ACTION_SEND
+        sendIntent.type = "text/plain"
+        sendIntent.putExtra(Intent.EXTRA_TEXT, EventUtils.getSharableInfo(event))
+        eventImage.tag?.let {
+            val bmpUri = EventUtils.getLocalBitmapUri(eventImage)
+            if (bmpUri != null) {
+                sendIntent.type = "image/*"
+                sendIntent.putExtra(Intent.EXTRA_STREAM, bmpUri)
+            }
+        }
+        eventImage.context.startActivity(Intent.createChooser(sendIntent, "Share Event Details"))
+    }
+
+    private fun getLocalBitmapUri(imageView: ImageView): Uri? {
+        val drawable = imageView.drawable
+        val bmp: Bitmap
```

Comment:
Use same line for declaration and assignment

Reply:
it'll be inaccessible in the try block.

Reply:
```val bmp = try {

} catch {
	null
}

if (bmp == null)
  return null
```

Reply:
```suggestion
      val drawable = imageView.drawable
        if (drawable is BitmapDrawable) {
            drawable.bitmap?.let {
                 val file = File(imageView.context.cacheDir, "share_image_" + System.currentTimeMillis() + ".png")
                 FileOutputStream(file)
                     .use { fos ->
                         it.compress(Bitmap.CompressFormat.PNG, 90, fos)
                         Timber.d("file: %s", file.absolutePath)
                         val bmpUri =
                             FileProvider.getUriForFile(imageView.context, BuildConfig.APPLICATION_ID + ".provider", file)
                         Timber.d("uri: %s", bmpUri)
                         return bmpUri
                     }
             }
        }
        return null
```
Better?

Reply:
```val drawable = imageView.drawable
        if (drawable is BitmapDrawable) {
            drawable.bitmap?.let {
                 val file = File(imageView.context.cacheDir, "share_image_" + System.currentTimeMillis() + ".png")
                 FileOutputStream(file)
                     .use { fos ->
                         it.compress(Bitmap.CompressFormat.PNG, 90, fos)
                         Timber.d("file: %s", file.absolutePath)
                         val bmpUri =
                             FileProvider.getUriForFile(imageView.context, BuildConfig.APPLICATION_ID + ".provider", file)
                         Timber.d("uri: %s", bmpUri)
                         return bmpUri
                     }
             }
        }
        return null```

Better?

Reply:
Invert the condition

```
if (drawable is not BitmapDrawable)
    return null

..continue
```

Reply:
Why?
We still have to return null in the end.

Reply:
We already are returning null after the condition check. Why do we still have to return null in the end?

Reply:
Because of the ?.let

Reply:
```
if (drawable is not BitmapDrawable)
    return null

..continue
```

Reply:
![Screenshot from 2019-04-08 15-09-29](https://user-images.githubusercontent.com/24875315/55714816-f94c1400-5a10-11e9-9d56-02aba2901248.png)

IDE suggestion
```
 val drawable = imageView.drawable as? BitmapDrawable ?: return null

        drawable.bitmap?.let {
            val file = File(imageView.context.cacheDir, "share_image_" + System.currentTimeMillis() + ".png")
            FileOutputStream(file)
                .use { fos ->
                    it.compress(Bitmap.CompressFormat.PNG, 90, fos)
                    Timber.d("file: %s", file.absolutePath)
                    val bmpUri =
                        FileProvider.getUriForFile(imageView.context, BuildConfig.APPLICATION_ID + ".provider", file)
                    Timber.d("uri: %s", bmpUri)
                    return bmpUri
                }
        }
        return null
```

Reply:
```
if (imageView.drawable as? BitmapDrawable == null || imageView.drawable.bitmap == null)
    return null

val file = File(imageView.context.cacheDir, "share_image_" + System.currentTimeMillis() + ".png")
return FileOutputStream(file)
  .use { fos ->
    it.compress(Bitmap.CompressFormat.PNG, 90, fos)
    Timber.d("file: %s", file.absolutePath)
    val bmpUri =
    FileProvider.getUriForFile(imageView.context, BuildConfig.APPLICATION_ID + ".provider", file)
   Timber.d("uri: %s", bmpUri)
   bmpUri
 }
```

Reply:
done..